docs/src/main/tut/chisel3/modules.md

```diff
@@ -84,3 +84,68 @@ children modules, using the ```Module``` constructor function and
 the Scala ```new``` keyword to create a
 new object.  We then wire them up to one another and to the ports of
 the ```Mux4``` interface.
+
+### RawModule
+
+*Module* are used to connect components inside a design. But once we
+want to synthesize for the real world (FPGA, ASIC, ...) some extra
+features are required like io specific naming, reset polarity, clock
+assignement, ...
+
+For this, a specific module named *RawModule* should be used. With RawModule
+there is no implicit signals. All signals like clock and reset should be
+given in IO argument.
+
+Then we can use it in place of *Module* usage :
+```tut:silent
+class SlaveSpi extends Module {
+  val io = IO(new Bundle{
+    val mosi = Input(Bool())
+    val miso = Output(Bool())
+    val sclk = Input(Bool())
+    val csn = Input(Bool())
+  })
+[...]
+}
+
+class TopSlaveSpi extends RawModule {
+  val clock = IO(Input(Clock()))
+  val rstn  = IO(Input(Bool()))
+  val mosi = IO(Input(Bool()))
+  val miso = IO(Output(Bool()))
+  val sclk = IO(Input(Bool()))
+  val csn = IO(Input(Bool()))
+
+  withClockAndReset(clock, !rstn) {
+    val slavespi = Module(new SlaveSpi)
+    miso := slavespi.io.miso
+    slavespi.io.mosi := mosi
+    slavespi.io.sclk := sclk
+    slavespi.io.csn  := csn
+  }
+}
+```
+
+In the example above, the RawModule is used to change the reset polarity
+of module SlaveSpi. Indeed, the reset is active high by default in chisel
+modules, then using withClockAndReset(clock, !rstn) we can use an active low
+reset in entire design.
+
+The clock is just wired as it, but if needed, RawModule can be used in
+conjuction with BlackBox to connect a differential clock input for example.
+
+With multi-lines IO() declaration we also rename signal in emitted verilog.
+Instead of *io_* named signals we will have a same names has val written:
```

Comment:
I think this documentation should likely be moved into the documentation for `MultiIOModule` if we introduce `Module` first, `MultiIOModule` second, and `RawModule` third.